js/src/name-pubsub/cancel.js

```diff
@@ -63,24 +53,35 @@ module.exports = (createCommon, options) => {
 
     it('should cancel a subscription correctly returning true', function (done) {
       this.timeout(300 * 1000)
-      const ipnsPath = `/ipns/${nodeId}`
-
-      series([
-        (cb) => ipfs.name.pubsub.subs(cb),
-        (cb) => ipfs.name.publish(value, { resolve: false }, cb),
-        (cb) => ipfs.name.resolve(nodeId, cb),
-        (cb) => ipfs.name.pubsub.subs(cb),
-        (cb) => ipfs.name.pubsub.cancel(ipnsPath, cb),
-        (cb) => ipfs.name.pubsub.subs(cb)
-      ], (err, res) => {
+
+      PeerId.create({ bits: 512 }, (err, peerId) => {
         expect(err).to.not.exist()
-        expect(res).to.exist()
-        expect(res[0]).to.eql([]) // initally empty
-        expect(res[4]).to.have.property('canceled')
-        expect(res[4].canceled).to.eql(true)
-        expect(res[5]).to.be.an('array').that.does.not.include(ipnsPath)
 
-        done()
+        const id = peerId.toB58String()
+        const ipnsPath = `/ipns/${id}`
+
+        ipfs.name.pubsub.subs((err, res) => {
+          expect(err).to.not.exist()
+          expect(res).to.be.an('array').that.does.not.include(ipnsPath)
+
+          ipfs.name.resolve(id, (err) => {
+            expect(err).to.exist()
+            auto({
+              subs1: (cb) => ipfs.name.pubsub.subs(cb),
+              cancel: (cb) => ipfs.name.pubsub.cancel(ipnsPath, cb),
+              subs2: (cb) => ipfs.name.pubsub.subs(cb)
+            }, 1, (err, res) => {
```

Comment:
With auto you need to specify your dependencies otherwise this will all run in parallel:

````js
            auto({
              subs1: (cb) => ipfs.name.pubsub.subs(cb),
              cancel: ['subs1', (_, cb) => ipfs.name.pubsub.cancel(ipnsPath, cb)],
              subs2: ['cancel', (_, cb) => ipfs.name.pubsub.subs(cb)]
            } /* ... */)
```